agent/src/com/jonnyzzz/teamcity/virtual/run/docker/DockerVM.java

```diff
@@ -88,49 +89,62 @@ public void buildWithScriptFile(@NotNull final File script) throws RunBuildExcep
         );
 
         builder.addTryProcess(
-                block("Executing the command using " + ctx.getShellLocation(), cmd.commandline(
+                block("Executing the command using " + ctx.getShellLocationInsideContainer(), cmd.commandline(
                         checkoutDir,
                         dockerRun(name, workDir, additionalCommands, scriptRun(script))
                 ))
         );
-        builder.addFinishProcess(block("Terminating images (if needed)", cmd.commandline(checkoutDir, Arrays.asList("docker", "kill", name, "2>&1", "||", "true"))));
 
-        builder.addFinishProcess(block("Fixing chown", new DelegatingBuildProcess(new DelegatingBuildProcess.ActionAdapter() {
-          @NotNull
-          @Override
-          public BuildProcess startImpl() throws RunBuildException {
-            final CompositeBuildProcess bp = new CompositeBuildProcessImpl();
-
-            final String uid = mySidAndGid.getUID();
-            final String gid = mySidAndGid.getGID();
-
-            if (StringUtil.isEmptyOrSpaces(uid) || StringUtil.isEmptyOrSpaces(gid)) {
-              logger.warning("SID and GID of current user were not found. chown is skipped");
-              return NOP;
+        List<String> terminatingCommand;
+        if (SystemInfo.isUnix) {
+          terminatingCommand = Arrays.asList("docker", "kill", name, "2>&1", "||", "true");
+        } else {
+          terminatingCommand = Arrays.asList("docker", "kill", name, ">nul", "2>nul", "&", "exit", "0");
```

Comment:
This approach swallows possible error output from stop command. It may turn out to be tricky to investigate build issues. Let's redirect STDERR to STDOUT instead.
Is it "&" and not "||"?